examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1468 @@
+    // following attributes are defined in the jupyter notebook, stored in
+    // `data` as HDF5 attributes and then read by the constructor.
+    const double     max_force_amplitude;
+    const double     force_sigma_x;
+    const double     force_sigma_y;
+    const double     max_force_width_x;
+    const double     max_force_width_y;
+    const Point<dim> force_center;
+
+  public:
+    // In this particular simulation the force has only a $x$ component,
+    // $F_y=0$.
+    const unsigned int force_component = 0;
+  };
+
+  // @sect4{`PML` class}
+  // This class is used to define the shape of the PML.
+  template <int dim>
+  class PML : public Function<dim, std::complex<double>>
+  {
+  public:
+    PML(HDF5::Group &data);
+    virtual std::complex<double> value(const Point<dim> & p,
+                                       const unsigned int component) const;
+
+  private:
+    // HDF5::Group in which all the simulation results will be stored.
+    HDF5::Group data;
+
+    // The same as before, the following attributes are defined in the jupyter
+    // notebook, stored in `data` as HDF5 attributes and then read by the
+    // constructor.
+    const double pml_coeff;
+    const int    pml_coeff_degree;
+    const double dimension_x;
+    const double dimension_y;
+    const bool   pml_x;
+    const bool   pml_y;
+    const double pml_width_x;
+    const double pml_width_y;
+    const double a_coeff_x;
+    const double a_coeff_y;
+  };
+
+
+
+  // @sect4{`Rho` class}
+  // This class is used to define the mass density.
+  template <int dim>
+  class Rho : public Function<dim>
+  {
+  public:
+    Rho(HDF5::Group &data);
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const;
+
+  private:
+    // HDF5::Group in which all the simulation results will be stored.
+    HDF5::Group data;
+
+    // The same as before, the following attributes are defined in the jupyter
+    // notebook, stored in `data` as HDF5 attributes and then read by the
+    // constructor.
+    const double       lambda;
+    const double       mu;
+    const double       material_a_rho;
+    const double       material_b_rho;
+    const double       cavity_resonance_frequency;
+    const unsigned int nb_mirror_pairs;
+    const double       dimension_y;
+    const unsigned int grid_level;
+    double             average_rho_width;
+  };
+
+
+
+  // @sect4{`Parameters` class}
+  // This class contains all the parameters that will be used in the simulation.
+  template <int dim>
+  class Parameters
+  {
+  public:
+    Parameters(HDF5::Group &data);
+
+    // HDF5::Group in which all the simulation results will be stored.
+    HDF5::Group data;
+
+    // The same as before, the following attributes are defined in the jupyter
+    // notebook, stored in `data` as HDF5 attributes and then read by the
+    // constructor.
+    const std::string        simulation_name;
+    bool                     save_vtu_files;
```

Comment:
It is a bit confusing why some of the member variables of this class are `const` and some are not. Is there a good reason?

Reply:
Now all the variables are const.